### PR TITLE
Pr/strings fixes

### PR DESCRIPTION
**Pull request by uli42**
fix some compiler warnings. This lead to rework string handling in Error.c: instead of using preallocated string arrays the strings are allocated using asprintf

### CODE REVIEW

File: nx-X11/programs/Xserver/hw/nxagent/Dialog.c [hunk 1]
```diff
@@ -315,13 +315,23 @@ void nxagentLaunchDialog(DialogType dialogType)
...
-    snprintf(dialogDisplay, sizeof(dialogDisplay), ":%s", display);
+    len = asprintf(&dialogDisplay, ":%s", display);
+  }
+
+  if (len == -1)
+  {
+    #ifdef DEBUG
+    fprintf(stderr, "%s: could not allocate display string.\n", __func__);
+    #endif
+    return;
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Dialog.c:321]: So... we simply ignore memory allocation errors here? Is that intended?

File: nx-X11/programs/Xserver/hw/nxagent/Error.c [hunk 2]
```diff
@@ -323,64 +323,53 @@ void nxagentEndRedirectToClientsLog(void)
   nxagentCloseClientsLogFile();
 }
 
+/*
+ * returns a pointer to the static nxagentHomeDir. The caller must not free
+ * this pointer!
+ */
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Error.c:326]: Where are you freeing nxagentHomeDir, finally, then? Leaving it to the application tidy up at end of application run?

File: nx-X11/programs/Xserver/hw/nxagent/Error.c [hunk 3]
```diff
@@ -372,68 +372,63 @@ char *nxagentGetHomePath(void)
   return nxagentHomeDir;
 }
 
+/*
+ * returns a pointer to the static nxagentRootDir. The caller must not free
+ * this pointer!
+ */
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Error.c:378]: Where do you free nxagentRootDir, finally? Leaving it to the application cleanup at end of runtime?

**Review by sunweaver — COMMENTED** [review]
Request for feedback on inline questions. The approach is good. I am wondering about some memory freeings.

File: nx-X11/programs/Xserver/hw/nxagent/Dialog.c (hunk 1 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Dialog.c:321]: well, what else are you suggesting?

File: nx-X11/programs/Xserver/hw/nxagent/Error.c (hunk 2 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Error.c:326]: well, nxagentHomedir is supposed to last during the nxagent runtime. Cleaning up such global variables does not have a place right now. Maybe ddxGiveUp() or AbortDDX(), not sure.

File: nx-X11/programs/Xserver/hw/nxagent/Error.c (hunk 3 above)
Comment (uli42) [reply nx-X11/programs/Xserver/hw/nxagent/Error.c:378]: yes, same for nxagentHomeDir.

**Comment by sunweaver** [comment]
On  So 05 Jan 2020 22:32:41 CET, Ulrich Sibiller wrote:

> uli42 commented on this pull request.
>
>
>
>> @@ -323,64 +323,53 @@ void nxagentEndRedirectToClientsLog(void)
>    nxagentCloseClientsLogFile();
>  }
>
> +/*
> + * returns a pointer to the static nxagentHomeDir. The caller must not free
> + * this pointer!
> + */
>
> well, nxagentHoemdir is supposed to last during the nxagent runtime.  
> Cleaning up such global variables does not have a place right now.  
> Maybe ddxGiveUp() or AbortDDX(), not sure.

hmmm... if we do it consciously, then let's maybe add a comment near  
the allocation of the var?

-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by uli42** [comment]
done

**Comment by sunweaver** [comment]
please rebase pr/strings_fixes, the rebase is not trivial.

**Comment by uli42** [comment]
done

File: nx-X11/programs/Xserver/hw/nxagent/Dialog.c (hunk 1 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Dialog.c:321]: hmmm... ok...

**Review by sunweaver — APPROVED** [review]
Happy with it now...

**Pull request merged**